pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -375,7 +375,15 @@ func (lb *ListenerBuilder) buildHTTPConnectionManager(httpOpts *httpListenerOpts
 		// Metadata exchange filter needs to be added before any other HTTP filters are added. This is done to
 		// ensure that mx filter comes before HTTP RBAC filter. This is related to https://github.com/istio/istio/issues/41066
 		if features.MetadataExchange && !httpOpts.hbone && !lb.node.IsAmbient() {
-			filters = append(filters, xdsfilters.HTTPMx)
+			if util.IsIstioVersionGE119(lb.node.IstioVersion) {
```

Comment:
I'd like use a pilot feature flags rather than base on version.

Reply:
Yeah, let me add an environment variable to disable it if necessary